### PR TITLE
DB update event, task and exception init

### PR DESCRIPTION
**Pull request by MDziugas**

### CODE REVIEW

File: CSE/Logic/DataManagement/Converter.cs [hunk 1]
```diff
@@ -5,20 +5,27 @@
...
 using Logic.Database;
 using System.Diagnostics;
 
 namespace Logic.DataManagement
 {
-    public class Converter : IConverter
+	public class Converter : IConverter
     {
-        private readonly ITextProcessing _textProcessing;
-        private readonly IUpdater _updater;
```
Comment (michaalx) [inline CSE/Logic/DataManagement/Converter.cs:16]: Why have you deleted IUpdater field?

File: CSE/Logic/Database/Reader.cs [hunk 2]
```diff
@@ -39,15 +43,19 @@ public IEnumerable<string> ReadProductData()
 			{
 				productList.Add(_reader.GetString(0).Trim());
 			}
+			Console.WriteLine("ReadProductData");
```
Comment (michaalx) [inline CSE/Logic/Database/Reader.cs]: Is this line needed here?

Comment (MDziugas) [reply CSE/Logic/Database/Reader.cs]: Definitely not

File: CSE/Logic/DataManagement/Converter.cs (hunk 1 above)
Comment (MDziugas) [reply CSE/Logic/DataManagement/Converter.cs:16]: Accident made when resolving conflicts, my bad

File: CSE/CSE.sln [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio 15
-VisualStudioVersion = 15.0.27004.2009
+VisualStudioVersion = 15.0.27004.2008
```
Comment (michaalx) [inline CSE/CSE.sln]: Can you update your VS to current version? I would hesitate downgrading this version.

Comment (MDziugas) [reply CSE/CSE.sln]: 15.0.27004.2009 is my current version, yet the one in the current master is 15.0.27004.2008 so I changed it back

File: CSE/Logic/DataManagement/TaskInit.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+		{
+			ThreadStart update = new ThreadStart(UpdateThread);
+			Thread thread = new Thread(update);
+		}
+
+		public void UpdateThread()
+		{
+
+			Store store = Store.Iki;
+			DataModel dm = new DataModel();
```
Comment (michaalx) [inline CSE/Logic/DataManagement/TaskInit.cs]: I see you don't use dependency injection here, not sure if after merging this pull request errors won't appear as parameterless constructor for DataModel does not exist as far as I know. (The same for Updater etc).

File: CSE/Logic/DataManagement/TaskInit.cs [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+using Logic.Database;
+using Logic.Metadata;
+using System.Threading;
+using Logic.Models;
+
+namespace Logic.DataManagement
+{
+	public class TaskInit
+	{
+		public Receipt receipt;
```
Comment (michaalx) [inline CSE/Logic/DataManagement/TaskInit.cs]: Do you really need two properties with same name (only case different)?

**Review by michaalx — APPROVED** [review]
Marvellous job!

**Pull request merged**